src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/WaitRequestHandler.kt

```diff
@@ -61,21 +57,43 @@ class WaitRequestHandler(private val config: FrejaEidAuthenticatorPluginConfig)
         const val SESSION_CUSTOM_IDENTIFIER = "freja-custom-identifier"
         const val SESSION_TIMESTAMP = "freja-timestamp"
         const val SESSION_INTEGRATOR_SPECIFIC_USER_ID = "freja-integrator-specific-user-id"
+        const val QR_CODE = "_qrCode"
+        const val CSP_OVERRIDE_IMG_SRC = "_cspImgsrc"
+        const val THIS_DEVICE_LINK = "_thisDeviceLink"
     }
 
     private val _logger: Logger = LoggerFactory.getLogger(StartRequestHandler::class.java)
 
     private val _json = config.json
     private val _sessionManager = config.sessionManager
     private val _relyingPartyId = config.relyingPartyId.orElse(null)
+    private val _requestLogicHelper = RequestLogicHelper(config)
 
     override fun get(requestModel: RequestModel, response: Response): Optional<AuthenticationResult> = Optional.empty()
+    //if config enables qr code render it, first calling to get the authRef
 
     override fun preProcess(request: Request, response: Response): RequestModel
     {
         if (request.isGetRequest)
         {
-            response.setResponseModel(ResponseModel.templateResponseModel(emptyMap(), "authenticate/wait"),
+            var viewData = emptyMap<String, String>()
+            if (config.qrCodeEnabled())
+            {
+                //the following data are proposed from freja documentation on QRCode generation
+                val postData = mutableMapOf("userInfoType" to "INFERRED", "userInfo" to "N/A")
+                val responseData = _requestLogicHelper.getAuthTransaction(postData)
+                val authRef = responseData["authRef"]?.toString()
+                config.sessionManager.put(Attribute.of(SESSION_AUTH_REF, authRef))
+                val baseUrl = if (config.environment == PredefinedEnvironment.PRODUCTION) QR_CODE_GENERATE_URL_PROD else QR_CODE_GENERATE_URL_TEST
```

Comment:
why not a ternary operator rather than if/else?

Reply:
I think that code was removed.